### ISSUE TITLE
Special characters ÄÖÜßñ are not transcribed properly

### ISSUE THREAD

**Issue by jmtatsch** [ggml-org/whisper.cpp#923]
When using talk-lama with -I de I noticed that German umlaute ÄÖÜß are not transcribed properly
```
main : done! start speaking in the microphone

Human: Der Fhrer fhrt berhaupt nicht gut.
LLaMA:
```

This should have been transcribed as:
`Human: Der Führer führt überhaupt nicht gut.`

This is probably of interest for most romance languages that have some special characters.

For Spanish the ñ is also missing:
`Human: Me gustan las jalapeos.`

Maybe there is UTF8 support missing somewhere?

**Comment by j0t4** [comment]
In windows , I can confirm the output is DOS850 code page.
You get :

"se├▒oras y se├▒ores"

instead of:

"señoras y señores"

"├▒"  (in DOS850)  ===   "ñ" (in utf8)


Thanks for your work.

**Comment by petterreinholdtsen** [comment]
I believe this is caused by this code line in talk-llama.cpp:

```
// remove all characters, except for letters, numbers, punctuation and ':', '\'', '-', ' '
text_heard = std::regex_replace(text_heard, std::regex("[^a-zA-Z0-9\\.,\\?!\\s\\:\\'\\-]"), "");
```

I suspect it would need to be extended with non-ascii letters to solve this issue.

**Comment by jmtatsch** [comment]
Yes, that is indeed the culprit. After commenting it out it works as expected.

If I understand correctly that regex is there to filter out unprintable characters.
Maybe that could be done more elegantly with something like` \p{C}` . I couldn't get it to work with c++ though

**Comment by petterreinholdtsen** [comment]
https://github.com/ggml-org/whisper.cpp/blob/master/examples/talk-llama/talk-llama.cpp still contain

```
// remove all characters, except for letters, numbers, punctuation and ':', '\'', '-', ' '
 text_heard = std::regex_replace(text_heard, std::regex("[^a-zA-Z0-9åäöÅÄÖ\\.,\\?!\\s\\:\\'\\-]"), "");
```
